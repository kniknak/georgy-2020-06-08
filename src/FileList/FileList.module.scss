.list {
    display: flex;
    flex-wrap: wrap;
    padding: 12px 6px;
    list-style: none;
}

