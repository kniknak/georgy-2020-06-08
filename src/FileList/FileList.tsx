import React from "react";
import { FileInfo } from "../api/types";
import { FileItem } from "./FileItem/FileItem";
import * as styles from "./FileList.module.scss";
import { FileListInfo } from "./FileListInfo/FileListInfo";

export interface FileListProps {
    files: FileInfo[]
    deleteFile: (id: string) => void
}

export const FileList: React.FC<FileListProps> = ({ files, deleteFile }) => {
    return (
        <div>
            <FileListInfo files={files}/>
            <ul className={styles.list}>
                {files.map(file => (
                    <FileItem
                        key={file._id}
                        file={file}
                        deleteFile={deleteFile}
                    />
                ))}
            </ul>
        </div>
    );
};
