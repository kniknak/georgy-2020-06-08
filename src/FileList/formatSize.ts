export const formatSize = (size: number) => (
    size >= 1024 * 1024
        ? Math.round(size / 1024 / 1024) + "Mb"
        : (
            size >= 1024
                ? Math.round(size / 1024) + "kb"
                : size + "b"
        )
);