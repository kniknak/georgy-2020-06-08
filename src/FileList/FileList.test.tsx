import React from "react";
import { shallow } from "enzyme";
import { FileList, FileListProps } from "./FileList";

describe("FileList", () => {
    describe("FileList", () => {
        const defaultProps: FileListProps = {
            files: [{
                originalName: "foo",
                size: 42,
                _id: "bar",
                createdAt: 200,
            }],
            deleteFile: () => undefined,
        };

        it("renders", () => {
            const subject = shallow(<FileList {...defaultProps}/>);

            expect(subject).toMatchSnapshot();
        });
    });
});