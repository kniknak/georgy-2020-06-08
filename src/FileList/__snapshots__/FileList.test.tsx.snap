// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList FileList renders 1`] = `
<div>
  <FileListInfo
    files={
      Array [
        Object {
          "_id": "bar",
          "createdAt": 200,
          "originalName": "foo",
          "size": 42,
        },
      ]
    }
  />
  <ul>
    <FileItem
      deleteFile={[Function]}
      file={
        Object {
          "_id": "bar",
          "createdAt": 200,
          "originalName": "foo",
          "size": 42,
        }
      }
      key="bar"
    />
  </ul>
</div>
`;
